components/backdrop/backdrop.js

```diff
@@ -0,0 +1,151 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { getComposedChildren, getComposedParent } from '../../helpers/dom.js';
+
+let count = 0;
+let overflow = null;
+
+class Backdrop extends LitElement {
+
+	static get properties() {
+		return {
+			show: { type: Boolean, reflect: true },
+			forTarget: { type: String, attribute: 'for-target' },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [ css`
+			:host {
+				background-color: var(--d2l-color-regolith);
+				height: 0;
+				left: 0;
+				opacity: 0;
+				position: fixed;
+				top: 0;
+				transition: opacity 200ms ease-in;
+				width: 0;
+				z-index: 999;
+			}
+			:host([_state="showing"]), :host([_state="hiding"]) {
+				height: 100%;
+				width: 100%;
+			}
+			:host([_state="showing"]) {
+				opacity: 0.7;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.show = false;
+		this._state = null;
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+
+		if (name === 'show' && oldval !== newval) {
```

Comment:
Nit: I don't think `attributeChangedCallback` will be called if `oldval === newval`.

Also nit: I'd invert this statement and return right away to reduce the nesting of the rest of the code.